packages/typegen/src/index.ts

```diff
@@ -124,34 +131,29 @@ export const generate = async (params: Partial<Options>) => {
   }
 
   const findAll = async () => {
-    const getColumnInfo = columnInfoGetter(pool)
+    const cwd = path.resolve(process.cwd(), rootDir)
+    const logMsgIgnore = undefined === ignore ? '' : ` ignoring ${ignore}`
+    const logMsgSince = undefined === since ? '' : ` since ${since}`
+    logger.info(`Matching files in ${getLogPath(cwd)} with pattern ${glob}${logMsgIgnore}${logMsgSince}`)
 
-    const globParams: Parameters<typeof globAsync> =
-      typeof glob === 'string'
-        ? [glob, {}]
-        : 'since' in glob
-        ? [globList(changedFiles({since: glob.since, cwd: path.resolve(rootDir)})), {}]
-        : glob
-
-    const getLogPath = (filepath: string) => {
-      const relPath = path.relative(process.cwd(), filepath)
-      return relPath.charAt(0) === '.' ? relPath : `./${relPath}`
-    }
+    const getColumnInfo = columnInfoGetter(pool)
 
     const getLogQueryReference = (query: {file: string; line: number}) => `${getLogPath(query.file)}:${query.line}`
 
-    const getFiles = () => {
-      logger.info(
-        `Searching for files matching ${globParams[0]} in ${getLogPath(path.resolve(process.cwd(), rootDir))}`,
-      )
-      return globAsync(globParams[0], {
-        ...globParams[1],
-        cwd: path.resolve(process.cwd(), rootDir),
+    const getFiles = async () => {
+      logger.info(`Searching for files.`)
+      let files = await globAsync(glob, {
+        ignore,
+        cwd,
         absolute: true,
-      }).then(files => {
-        logger.info(`Found ${files.length} files matching pattern.`)
-        return files
       })
+      if (since !== undefined) {
```

Comment:
Again I would disagree with this change, because I didn't write it like this because I think since might be an empty string; I wrote it like this because I want to convey to the reader of the code that I'm checking for `undefined`.

If a consumer passes in an empty string for `since` or `ignore`, who am I to say "Oh silly old you! What you probably meant is `undefined`! Let me fix that for you."

But I already made my point about the differentiation between `nullish` vs `truthy` and how this often comes down to preference. And as the maintainer the preference that counts here is yours. :)